### PR TITLE
New geotools package

### PR DESCRIPTION
**Pull request by mdastous-bentley**

### CODE REVIEW

**Review by calebmshafer — COMMENTED** [review]
Looks good overall, I didn't take that deep of a look at the react part of it but I'll dig into it more tomorrow.

**Comment by calebmshafer** [comment]
@mdastous-bentley mind signing the CLA? The link is above.  Thanks!

File: packages/geo-tools/src/components/GeoAddressSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        }}
+        {...otherProps}
+      />
+    );
+  }), []);
+
+  return (
+    <div className="geotools-geoaddresssearch__container">
+      <div className="geotools-geoaddresssearch__autosuggest">
+        <AutoSuggest
```
Comment (aruniverse) [inline packages/geo-tools/src/components/GeoAddressSearch.tsx:83]: I was unaware of the component from the ui-core lib :)

File: packages/geo-tools/src/GeoToolsItemDef.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    return new CustomItemDef({
+      customId: "geo-tools:geoAddressSearch",
+      iconSpec: "icon-search",
+      labelKey: "GeoTools:geoAddressSearch.label",
+      popupPanelNode: <ToolbarPopupContext.Consumer>
+        {() => (
+          <GeoAddressSearch />
+        )}
+      </ToolbarPopupContext.Consumer>,
+      // DEPRECATED way (still used by DR)
```
Comment (aruniverse) [inline packages/geo-tools/src/GeoToolsItemDef.tsx:32]: I will note this down, and make sure DR gets updated accordingly, thx

File: packages/geo-tools/src/IModelGeoView.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    const range = new CartographicRange(viewRange, ecef.getTransform());
+
+    return range.getLongitudeLatitudeBoundingBox();
+  }
+
+  public static locateAddress(address: string): boolean {
+
+    const vp = IModelApp.viewManager?.selectedView;
+    if (vp === undefined || vp.iModel === undefined || vp.iModel.contextId === undefined || vp.iModel.iModelId === undefined)
```
Comment (aruniverse) [inline packages/geo-tools/src/IModelGeoView.ts:39]: nit. could use optional chaining here to clean this up a bit

Comment (aruniverse) [reply packages/geo-tools/src/IModelGeoView.ts:39]: line 16 too

**Review by aruniverse — CHANGES_REQUESTED** [review]
Overall looks good. 

I'd say remove the Changelog json&md file in the geo-tools dir, as rush should auto generate that for you (i believe)

Please update the repo root level readme by adding this pkg to the list of pkgs in this monorepo. 
Please add yourself ass the codewoner in the root/.github/codeowner file

The main changes im requesting is in the pkg.json with the deps.

File: packages/geo-tools/src/IModelGeoView.ts (hunk 3 above)
Comment (mdastous-bentley) [reply packages/geo-tools/src/IModelGeoView.ts:39]: ok done.

File: packages/geo-tools/src/components/GeoAddressSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+export interface GeoAddressSearchProps extends CommonProps {
+  provider?: AddressProvider;
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function GeoAddressSearch(props: GeoAddressSearchProps) {
+  const [inputValue, setInputValue] = React.useState("");
+
+  // `React.useMemo' is used avoid creating new object on each render cycle
+  const addressProvider = React.useMemo(() => (props.provider ? props.provider : new BingAddressProvider()), [props.provider]);
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/geo-tools/src/components/GeoAddressSearch.tsx:24]: didnt know that, thanks !

File: packages/geo-tools/package.json [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    "jsdom": "^11.12.0",
+    "jsdom-global": "3.0.2",
+    "mocha": "^5.2.0",
+    "nyc": "^14.0.0",
+    "raf": "^3.4.0",
+    "rimraf": "^3.0.2",
+    "sinon": "^9.0.2",
+    "sinon-chai": "^3.2.0",
+    "source-map-support": "^0.5.6",
+    "tslint": "^5.11.0",
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/geo-tools/package.json:79]: done.

File: packages/geo-tools/package.json [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    "typescript": "~3.7.4",
+    "xmlhttprequest": "^1.8.0"
+  },
+  "peerDependencies": {
+    "@bentley/bentleyjs-core": "2.10.0-dev.5",
+    "@bentley/ui-framework": "2.10.0-dev.5",
+    "@bentley/imodeljs-common": "2.10.0-dev.5",
+    "@bentley/ui-components": "2.10.0-dev.5",
+    "@bentley/ui-core": "2.10.0-dev.5",
+    "@bentley/imodeljs-i18n": "^2.10.0-dev.5",
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/geo-tools/package.json:91]: make sense, anyway Im waiting 2.9.0 to be released.

File: packages/geo-tools/package.json [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    "pseudolocalize": "betools pseudolocalize --englishDir ./public/locales/en --out ./lib/public/locales/en-PSEUDO",
+    "test": "mocha --opts ./mocha.opts \"./lib/test/**/*.test.js\""
+  },
+  "dependencies": {
+    "i18next": "^10.2.2",
+    "classnames": "^2.2.5",
+    "react": "^16.8.0",
+    "react-dom": "^16.8.0",
+    "react-autosuggest": "^10.0.2",
+    "@bentley/itwin-client": "2.10.0-dev.5"
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/geo-tools/package.json:41]: So if its part of peer, it also needs to be part of dev?

File: packages/geo-tools/src/AddressProvider.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      degreeRange.extendXY(Angle.radiansToDegrees(point.x), Angle.radiansToDegrees(point.y));
+    });
+
+    const degrees = [degreeRange.low.y, degreeRange.low.x, degreeRange.high.y, degreeRange.high.x];
+
+    return `http://dev.virtualearth.net/REST/v1/Autosuggest?query=${query}&userMapView=${degrees}&maxResults=${this._maxResults}&includeEntityTypes=${entityTypesStr}&key=${this._bingKey}`;
+  }
+
+  /**
+ * longitude(X) are expressed in any range between -2PI to +2PI
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/geo-tools/src/AddressProvider.ts:51]: Not too sure what is wrong here?

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -24,3 +24,5 @@
 /packages/imodel-content-tree  @grigasp
 
 /packages/imodel-react-hooks @michaelbelousov
+
+/packages/geo-tools @mdastous-bentley
```
Comment (aruniverse) [inline .github/CODEOWNERS:28]: could you add yourself on line 14 for /common/changes

File: packages/geo-tools/package.json [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+  "dependencies": {
+    "classnames": "^2.2.5",
+    "i18next": "^10.2.2",
+    "react-autosuggest": "^10.0.2"
+  },
+  "devDependencies": {
+    "@bentley/build-tools": "2.10.0-dev.5",
+    "@bentley/eslint-plugin": "2.10.0-dev.5",
+    "@bentley/itwin-client": "2.10.0-dev.5",
+    "@bentley/telemetry-client": "2.10.0-dev.5",
```
Comment (aruniverse) [inline packages/geo-tools/package.json:44]: missing quite a few dev deps, any of the core imjs peer deps should be listed here as dev deps too

Comment (aruniverse) [reply packages/geo-tools/package.json:44]: i dont think you need telemetry-lcient, so can remove that. need to add geom-core, bentleyjs-core, ui-abstract, ui-frmework, ui-core, imjs-common, imjs-frontend, imjs-i18n

**Review by aruniverse — APPROVED** [review]

**Pull request merged**